PathfinderAPI/Util/XML/ElementInfo.cs

```diff
@@ -28,60 +45,222 @@ public override string ToString()
         {
             WriteToXML(writer);
         }
-
         return builder.Replace("\t", "  ").ToString();
     }
 
+    public bool TryContentAsBoolean(out bool result)
+        => Content.TryAsBoolean(out result);
+
+    public bool TryContentAsInt(out int result)
+        => Content.TryAsInt(out result);
+
+    public bool TryContentAsFloat(out float result)
+        => Content.TryAsFloat(out result);
+
+    public bool GetContentAsBoolean(bool defaultVal = default)
+        => TryContentAsBoolean(out var result) ? result : defaultVal;
+
+    public int GetContentAsInt(int defaultVal = default)
+        => TryContentAsInt(out var result) ? result : defaultVal;
+
+    public float GetContentAsFloat(float defaultVal = default)
+        => TryContentAsFloat(out var result) ? result : defaultVal;
+
     public bool ContentAsBoolean()
-        => bool.TryParse(Content, out var value)
-            ? value
-            : throw new FormatException($"Value of '{Name}' is not true or false");
+        => Content.AsBoolean(nameof(Content));
 
     public int ContentAsInt()
-        => int.TryParse(Content, out var value)
-            ? value
-            : throw new FormatException($"Value of '{Name}' is not an integer, e.g.: 0, 1, 2");
-        
+        => Content.AsInt(nameof(Content));
+
     public float ContentAsFloat()
-        => float.TryParse(Content, out var value)
-            ? value
-            : throw new FormatException($"Value of '{Name}' is not a float, e.g.: 1.0");
+        => Content.AsFloat(nameof(Content));
+
+    public bool TryAttributeAsBoolean(string attribName, out bool result)
```

Comment:
This naming scheme seems confusing, the behavior difference between GetAttributeAsX/AttributeAsX/TryAttributeAsX are not clear and I believe they would be prone to incorrect usage

Reply:
Yeah I probably should remove AttributeAs methods and instead add required and default behavior to GetAttributeAs like `GetAttributeAsInt(string name, bool required = false, int def = default)`